src/Microsoft.SymbolStore/KeyGenerators/PEFileKeyGenerator.cs

```diff
@@ -120,79 +128,94 @@ public override IEnumerable<SymbolStoreKey> GetKeys(KeyTypeFlags flags)
             }
         }
 
-        private IEnumerable<string> GetSpecialFiles(KeyTypeFlags flags)
+        private IEnumerable<string> GetSOSFiles(string runtimeFileName)
         {
-            List<string> specialFiles = new((flags & KeyTypeFlags.ClrKeys) != 0 ? s_coreClrSpecialFiles : s_dacdbiSpecialFiles);
+            if (runtimeFileName == ClrFileName)
+            {
+                return GetFilesLongNameVariants(SosFileName);
+            }
+
+            return Enumerable.Empty<string>();
+        }
+
+        private IEnumerable<string> GetDACFiles(string runtimeFileName)
+        {
+            if (runtimeFileName == CoreClrFileName)
+            {
+                string[] coreClrDACFiles = new string[] { CoreClrDACFileName, DbiFileName };
+                IEnumerable<string> longNameDACFiles = GetFilesLongNameVariants(CoreClrDACFileName);
+                return coreClrDACFiles.Concat(longNameDACFiles);
+            }
 
-            VsFixedFileInfo fileVersion = _peFile.VersionInfo;
-            if (fileVersion != null)
+            if (runtimeFileName == ClrFileName)
             {
-                ushort major = fileVersion.ProductVersionMajor;
-                ushort minor = fileVersion.ProductVersionMinor;
-                ushort build = fileVersion.ProductVersionBuild;
-                ushort revision = fileVersion.ProductVersionRevision;
+                string[] clrDACFiles = new string[] { ClrDACFileName, DbiFileName };
+                IEnumerable<string> longNameDACFiles = GetFilesLongNameVariants(ClrDACFileName);
+                return clrDACFiles.Concat(longNameDACFiles);
+            }
 
-                List<string> hostArchitectures = new();
-                string targetArchitecture = null;
+            return Enumerable.Empty<string>();
+        }
 
-                ImageFileMachine machine = (ImageFileMachine)_peFile.FileHeader.Machine;
-                switch (machine)
+        private IEnumerable<string> GetFilesLongNameVariants(params string[] filesWithLongNameVariant)
```

Comment:
Does this get used with more that one string at a time?

Reply:
Good catch, I'll change it back to accept a single string, now that its more clear what files we generate the long name variant for (just sos and dac, which are now generated in separate steps)